presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoConnection.java

```diff
@@ -106,8 +106,9 @@
         requireNonNull(uri, "uri is null");
         this.jdbcUri = uri.getJdbcUri();
         this.httpUri = uri.getHttpUri();
-        this.schema.set(uri.getSchema());
-        this.catalog.set(uri.getCatalog());
+        uri.getSchema().ifPresent(value -> schema.set(value));
+        uri.getCatalog().ifPresent(value -> catalog.set(value));
+        this.catalog.set(uri.getCatalog().orElse(null));
```

Comment:
"initialize" them also when uri's provided is empty:

```
this.schema.set(uri.getSchema().orElse(null))
...
```

also, catalog is set twice

Reply:
AtomicReference is null by default